hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/helpers/OmGetKey.java

```diff
@@ -0,0 +1,175 @@
+ * limitations under the License.
+ *
+ */
+package org.apache.hadoop.ozone.om.helpers;
+
+import org.apache.hadoop.ozone.om.OMMetadataManager;
+import org.apache.hadoop.ozone.om.request.file.OMFileRequest;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Iterator;
+
+/**
+ * This class help to get metadata keys.
+ */
+public final class OmGetKey {
+
+  private String volumeName;
+  private String bucketName;
+  private String keyName;
+  private OMMetadataManager omMetadataManager;
+  private long clientID;
+
+  private String fileName;
+  private Iterator<Path> pathComponents;
+  private long volumeId;
+  private long bucketId;
+  private long parentID;
+
+
+  @SuppressWarnings("checkstyle:parameternumber")
+  private OmGetKey(String volumeName, String bucketName, String keyName, 
+      OMMetadataManager omMetadataManager, long clientID,
+      String fileName, Iterator<Path> pathComponents, long volumeId,
+      long bucketId, long parentID) {
+    this.volumeName = volumeName;
+    this.bucketName = bucketName;
+    this.keyName = keyName;
+    this.omMetadataManager = omMetadataManager;
+    this.clientID = clientID;
+
+    this.fileName = fileName;
+    this.pathComponents = pathComponents;
+    this.volumeId = volumeId;
+    this.bucketId = bucketId;
+    this.parentID = parentID;
+  }
+
+  /**
+   * Builder class for OmGetKey.
+   */
+  public static class Builder {
+    private String volumeName;
+    private String bucketName;
+    private String keyName;
+    private OMMetadataManager omMetadataManager;
+    private long clientID;
+    private String errMsg;
+
+    public Builder() {
+      this.errMsg = null;
+    }
+
+    public Builder setVolumeName(String volumeName) {
+      this.volumeName = volumeName;
+      return this;
+    }
+
+    public Builder setBucketName(String bucketName) {
+      this.bucketName = bucketName;
+      return this;
+    }
+
+    public Builder setKeyName(String keyName) {
+      this.keyName = keyName;
+      return this;
+    }
+
+    public Builder setOmMetadataManager(OMMetadataManager omMetadataManager) {
+      this.omMetadataManager = omMetadataManager;
+      return this;
+    }
+
+    public Builder setClientID(long clientID) {
+      this.clientID = clientID;
+      return this;
+    }
+
+    public Builder setErrMsg(String errMsg) {
+      this.errMsg = errMsg;
+      return this;
+    }
+
+    public OmGetKey build() throws IOException {
+      String fileName = OzoneFSUtils.getFileName(this.keyName);
+      Iterator<Path> pathComponents = Paths.get(this.keyName).iterator();
+      final long volumeId = omMetadataManager.getVolumeId(this.volumeName);
+      final long bucketId = omMetadataManager.getBucketId(this.volumeName,
+          this.bucketName);
+      long parentID = OMFileRequest
+          .getParentID(volumeId, bucketId, pathComponents, this.keyName,
+          this.omMetadataManager, this.errMsg);
```

Comment:
This can be addressed in another Jira ticket, but from what I see since the `pathComponents` is derived from the `keyName`. We can remove this in `pathComponents` from the `OMFileRequest#getParentID` method arguments and instead call the   `Iterator<Path> pathComponents = Paths.get(this.keyName).iterator();` inside the `OMFileRequest#getParentID` instead.

I did some rough work on this: https://github.com/ivandika3/ozone/commit/4803bb27faa4b7790763d9952520fa62fd057d39#diff-4ea3716be41b5b8969a751fdc62db55f498b23ddb6f4d1085605dc974057dd1e

Reply:
@TaiJuWu I have created https://issues.apache.org/jira/browse/HDDS-9946 and assigned it to you. Please let me know if you have any questions.

Reply:
> @TaiJuWu I have created https://issues.apache.org/jira/browse/HDDS-9946 and assigned it to you. Please let me know if you have any questions.

Thanks, I will start working this weekend.